relay-common/src/metrics.rs

```diff
@@ -140,15 +177,51 @@ pub fn configure_statsd<A: ToSocketAddrs>(
     prefix: &str,
     host: A,
     default_tags: BTreeMap<String, String>,
+    buffering: bool,
+    sample_rate: f32,
 ) {
     let addrs: Vec<_> = host.to_socket_addrs().unwrap().collect();
     if !addrs.is_empty() {
         log::info!("reporting metrics to statsd at {}", addrs[0]);
     }
-    let statsd_client = StatsdClient::from_udp_host(prefix, &addrs[..]).unwrap();
+
+    let normalized_sample_rate = if sample_rate >= 1.0 {
+        1.0
+    } else if sample_rate <= 0.0 {
+        0.0
+    } else {
+        sample_rate
+    };
```

Comment:
_nit_: You can rewrite this as 

```rust
let sample_rate = sample_rate.max(0.).min(1.);
```

Note how I'm shadowing the input parameter to avoid it accidentally being used in place of the normalized value.